File: be_camphub/src/main/java/org/camphub/be_camphub/dto/response/dispute/DisputeResponse.java
package org.camphub.be_camphub.dto.response.dispute;

import java.util.List;
import java.util.UUID;

import org.camphub.be_camphub.dto.response.MediaResourceResponse;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class DisputeResponse {
    UUID id;
    UUID bookingId;
    UUID reporterId;
    UUID defenderId;
    UUID adminId;

    String reporterName; // tên người báo cáo
    String defenderName; // tên người bị báo cáo
    String adminName; // tên quản trị viên xử lý

    String description;
    List<MediaResourceResponse> evidenceUrls;
    String damageTypeName; // tên loại thiệt hại
    String damageTypeId;
    Double compensationRate;
    String status;
    String adminDecision;
    Double compensationAmount;
    String adminNote;

    String createdAt;
    String resolvedAt;
}

File: fe_camphub/src/libs/core/dto/response/dispute.response.ts
import { DisputeDecision, DisputeStatus } from "@/libs/core/constants";
import { MediaResource } from "../../types";

export interface DisputeResponse {
    id: string;                 // UUID
    bookingId: string;          // UUID
    reporterId: string;         // UUID
    defenderId: string;         // UUID
    adminId: string | null;     // backend có thể null

    reporterName: string;      // fullname của người báo cáo
    defenderName: string;      // fullname của người bị báo cáo
    adminName: string | null;  // fullname của admin xử lý

    description: string;
    evidenceUrls: MediaResource[];
    damageTypeId: string;
    damageTypeName: string;
    compensationRate: number | null;
    status: DisputeStatus;
    adminDecision: DisputeDecision;
    compensationAmount: number | null;
    adminNote: string;

    createdAt: string;          // ISO datetime
    resolvedAt: string | null;  // ISO datetime
}
